elastic/kibana
# [Watcher] Migrate from EuiCodeEditor to Monaco

Partially addresses: https://github.com/elastic/kibana/issues/107215

## Summary

This PR removes the dependency it has on ace editor in favour of Monaco. Note that `CodeEditor` is a kibana wide abstraction built on top of Monaco and located at `@kbn/kibana-react-plugin/public`.

#### Migrated to `Monaco`
- `application/sections/watch_edit/components/json_watch_edit/json_watch_edit_form.tsx`
- `application/sections/watch_edit/components/json_watch_edit/json_watch_edit_simulate.tsx`
- `application/sections/watch_edit/components/threshold_watch_edit/action_fields/webhook_action_fields.tsx`

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial -E path.repo=/tmp/es-backups` and kibana with `yarn start`
2. Navigate to `Stack Management` -> `Watcher`
3. Verify
  * Create advanced watch JSON editor (both in create form and in simulate )must be Monaco based, validation should work as before
  * Create threshold alert, the webhook action has a JSON editor that should now be Monaco based too.

<details>
<summary>Screenshots</summary>

![Screenshot 2023-04-21 at 11 02 37](https://user-images.githubusercontent.com/1191206/233597805-22aad4fb-981e-4302-afe3-766511ebbd08.png)
![Screenshot 2023-04-21 at 11 02 23](https://user-images.githubusercontent.com/1191206/233597815-99d0d499-762e-4cd2-a2c2-e1ab015ab3cf.png)
![Screenshot 2023-04-21 at 11 02 15](https://user-images.githubusercontent.com/1191206/233597819-b608b469-ddf1-41ca-9a4e-7b1c3b497e55.png)


</details>
